services/core-api/app/api/utils/setup_marshmallow.py

```diff
@@ -47,17 +75,39 @@ class Meta(object):
 
                     schema_class_name = "%sSchema" % class_.__name__
 
-                    schema_class = type(schema_class_name, (ModelSchema, ), {"Meta": Meta})
+                    class ModelSchema2(ModelSchema):
+                        activity_type_code = fields.String(dump_only=True)
+
+                    schema_class = type(schema_class_name, (ModelSchema2, ), {"Meta": Meta})
 
                     setattr(class_, "_schema", schema_class)
                     print(f'created {class_}')
                 except Exception as e:
                     raise e
 
+        for class_ in Base._decl_class_registry.values():
+            if hasattr(class_, "_schema"):
+                pprint(class_.__name__)
+                try:
+                    mapper = inspect(class_)
+
+                    for rel in mapper.relationships:
+                        pprint(rel.foreign_keys)
+                        if rel.property:
+                            pprint('  ' + rel.key)
+
+                            #raise Exception()
+                            class_._schema._declared_fields[rel.key] = fields.Nested(
+                                rel.entity.class_._schema, many=rel.uselist, dump_only=True)
+                            #exclude=[rel.backref.key] + [pk.name for pk in mapper.primary_keys])
+                except Exception as e:
+                    raise e
+
     return setup_schema_fn
 
 
 # TODO: finish this and resolve errors now_application/activity_detail_base.activity_type_code to all for programatic generation of schema
 # TODO: add call to model method to execute post_generation of schema.
-# event.listen(mapper, "after_configured", setup_schema(Base, db.session))
+
+#event.listen(mapper, "after_configured", setup_schema(BaseModel, db.session))
```

Comment:
if this line was uncommented, this would run, for now. it's just the start of future code.